Improve object documentation in lua_api.txt (#13239)
minetest/minetest: doc/lua_api.txt

### Methods

> * `get_pos()`: returns position in form `{x=num, y=num, z=num}`

Reviewer: Sounds kinda like vector metatable wasn't set.
```suggestion
* `get_pos()`: returns position as vector `{x=num, y=num, z=num}`
```

* `set_pos(pos)`:

>     * sets position in form `{x=num, y=num, z=num}`
>     * no-op if object is attached

Reviewer: ```suggestion
    * Sets the position of the object.
    * No-op if object is attached.
    * `pos` is a vector `{x=num, y=num, z=num}`
```

* `get_velocity()`: returns the velocity, a vector.
* `add_velocity(vel)`

>     * changes velocity by adding to the current velocity.

Reviewer: Most lines here use proper english sentences.
```suggestion
    * Changes velocity by adding to the current velocity.
```
(Also in the lines below.)

    * If `continuous` is true, the Lua entity will not be moved to the current
      position before starting the interpolated move.
    * For players this does the same as `set_pos`,`continuous` is ignored.
    * no-op if object is attached
* `punch(puncher, time_from_last_punch, tool_capabilities, direction)`
    * punches the object, triggering all consequences 'normal' punch would do
    * `puncher`: another `ObjectRef` which 'punched' the object,
    * `time_from_last_punch`: time since last punch action of the puncher

>     * `direction`: direction the punch came from (as a vector); can be `nil`

Reviewer: It would be better to just reference `on_punch` here, imo.

    * If `continuous` is true, the Lua entity will not be moved to the current
      position before starting the interpolated move.
    * For players this does the same as `set_pos`,`continuous` is ignored.
    * no-op if object is attached
* `punch(puncher, time_from_last_punch, tool_capabilities, dir)`

>     * punches the object, triggering all consequences 'normal' punch would do

Reviewer 2: ```suggestion
    * punches the object, triggering all consequences a 'normal' punch would have
```

>     * `puncher`: another `ObjectRef` which 'punched' the object

Reviewer 2: ```suggestion
    * `puncher`: another `ObjectRef` which punched the object
```

>     * Other arguments: See `on_punch` for entities

Reviewer 2: not mentioned that dir can be nil (it is automatically calculated IIRC)

* `right_click(clicker)`:
    * simulates using the 'place/use' key on the object
    * triggers all consequences as if a real player had done this

>     * `clicker` is another `ObjectRef` that has 'clicked'

Reviewer 2: ```suggestion
    * `clicker` is another `ObjectRef` which has clicked
```

* `get_inventory()`: returns an `InvRef` for players, otherwise returns `nil`
* `get_wield_list()`: returns the name of the inventory list the wielded item
   is in.

> * `get_wield_index()`: returns the hotbar index of the wielded item (starting with 1)

Reviewer: The hotbar is a hud thing, maybe better:
```suggestion
* `get_wield_index()`: returns the wield list index of the wielded item (starting with 1)
```

> * `get_wield_index()`: returns the wield list index of the wielded item (starting with 1)

Reviewer 2: aren't inventory lists always 1-indexed?

    * Attaches object to `parent`
    * See 'Attachments' section for details
    * `parent`: `ObjectRef` to attach to
    * `bone`: Bone to attach to. Default is `""` (the root bone)
    * `position`: relative position, default `{x=0, y=0, z=0}`
    * `rotation`: relative rotation in degrees, default `{x=0, y=0, z=0}`
    * `forced_visible`: Boolean to control whether the attached entity
       should appear in first person, default `false`.
    * This command may fail silently (do nothing) when it would result
      in circular attachments.
* `get_attach()`:
    * returns current attachment parameters or nil if it isn't attached
    * If attached, returns `parent`, `bone`, `position`, `rotation`, `forced_visible`
* `get_children()`: returns a list of ObjectRefs that are attached to the
    object.
* `set_detach()`: Detaches object. No-op if object was not attached.
* `set_bone_position([bone, position, rotation])`
    * `bone`: string. Default is `""`, the root bone
    * `position`: `{x=num, y=num, z=num}`, relative, `default {x=0, y=0, z=0}`
    * `rotation`: `{x=num, y=num, z=num}`, default `{x=0, y=0, z=0}`
* `get_bone_position(bone)`:

>     * returns `position, rotation` of the specified bone (as vectors)

Reviewer 2: this will only return values set before by `set_bone_position`, *not* actual model positions

* `get_inventory()`: returns an `InvRef` for players, otherwise returns `nil`
* `get_wield_list()`: returns the name of the inventory list the wielded item
   is in.
* `get_wield_index()`: returns the wield list index of the wielded item (starting with 1)
* `get_wielded_item()`: returns the wielded item as an `ItemStack`
* `set_wielded_item(item)`: replaces the wielded item, returns `true` if
  successful.
* `get_armor_groups()`:

>     * returns a table with all the objects's armor group ratings

Reviewer 3: ```suggestion
    * returns a table with all the object's armor group ratings
```

        * Fifth column:  subject viewed from above
        * Sixth column:  subject viewed from below
* `get_entity_name()` (**Deprecated**: Will be removed in a future version, use the field `self.name` instead)

> * `get_luaentity()`: returns the object's associated Lua entity

Reviewer: ```suggestion
* `get_luaentity()`: returns the object's associated luaentity table
```
